File: support_tools/swigwin-3.0.12/Examples/test-suite/ruby/overload_simple_runme.rb
#!/usr/bin/env ruby
#
# Put description here
#
# 
# 
# 
#

require 'swig_assert'

require 'overload_simple'

include Overload_simple

if foo(3) != "foo:int"
  raise RuntimeError, "foo(int)"
end

if foo(3.0) != "foo:double"
  raise RuntimeError, "foo(double)"
end

if foo("hello") != "foo:char *"
  raise RuntimeError, "foo(char *)"
end

f = Foo.new
b = Bar.new

if foo(f) != "foo:Foo *"
  raise RuntimeError, "foo(Foo *)"
end

if foo(b) != "foo:Bar *"
  raise RuntimeError, "foo(Bar *)"
end

v = malloc_void(32)

if foo(v) != "foo:void *"
  raise RuntimeError, "foo(void *)"
end

s = Spam.new

if s.foo(3) != "foo:int"
  raise RuntimeError, "Spam::foo(int)"
end

if s.foo(3.0) != "foo:double"
  raise RuntimeError, "Spam::foo(double)"
end

if s.foo("hello") != "foo:char *"
  raise RuntimeError, "Spam::foo(char *)"
end

if s.foo(f) != "foo:Foo *"
  raise RuntimeError, "Spam::foo(Foo *)"
end

if s.foo(b) != "foo:Bar *"
  raise RuntimeError, "Spam::foo(Bar *)"
end

if s.foo(v) != "foo:void *"
  raise RuntimeError, "Spam::foo(void *)"
end

if Spam.bar(3) != "bar:int"
  raise RuntimeError, "Spam::bar(int)"
end

if Spam.bar(3.0) != "bar:double"
  raise RuntimeError, "Spam::bar(double)"
end

if Spam.bar("hello") != "bar:char *"
  raise RuntimeError, "Spam::bar(char *)"
end

if Spam.bar(f) != "bar:Foo *"
  raise RuntimeError, "Spam::bar(Foo *)"
end

if Spam.bar(b) != "bar:Bar *"
  raise RuntimeError, "Spam::bar(Bar *)"
end

if Spam.bar(v) != "bar:void *"
  raise RuntimeError, "Spam::bar(void *)"
end

# Test constructors

s = Spam.new
if s.type != "none"
  raise RuntimeError, "Spam()"
end

s = Spam.new(3)
if s.type != "int"
  raise RuntimeError, "Spam(int)"
end
    
s = Spam.new(3.4)
if s.type != "double"
  raise RuntimeError, "Spam(double)"
end

s = Spam.new("hello")
if s.type != "char *"
  raise RuntimeError, "Spam(char *)"
end

s = Spam.new(f)
if s.type != "Foo *"
  raise RuntimeError, "Spam(Foo *)"
end

s = Spam.new(b)
if s.type != "Bar *"
  raise RuntimeError, "Spam(Bar *)"
end

s = Spam.new(v)
if s.type != "void *"
  raise RuntimeError, "Spam(void *)"
end

File: support_tools/swigwin-3.0.12/Examples/test-suite/lua/overload_simple_runme.lua
require("import")	-- the import fn
import("overload_simple")	-- import code
for k,v in pairs(overload_simple) do _G[k]=v end -- move to global

-- lua has only one numeric type, foo(int) and foo(double) are the same
-- whichever one was wrapper first will be used

assert(foo(3)=="foo:int" or foo(3)=="foo:double") -- could be either
assert(foo("hello")=="foo:char *")

f=Foo()
b=Bar()

assert(foo(f)=="foo:Foo *")
assert(foo(b)=="foo:Bar *")

v = malloc_void(32)

assert(foo(v) == "foo:void *")

s = Spam()

assert(s:foo(3) == "foo:int" or s:foo(3.0) == "foo:double") -- could be either
assert(s:foo("hello") == "foo:char *")
assert(s:foo(f) == "foo:Foo *")
assert(s:foo(b) == "foo:Bar *")
assert(s:foo(v) == "foo:void *")

assert(Spam_bar(3) == "bar:int" or Spam_bar(3.0) == "bar:double")
assert(Spam_bar("hello") == "bar:char *")
assert(Spam_bar(f) == "bar:Foo *")
assert(Spam_bar(b) == "bar:Bar *")
assert(Spam_bar(v) == "bar:void *")

-- Test constructors

s = Spam()
assert(s.type == "none")

s = Spam(3)
assert(s.type == "int" or s.type == "double")

s = Spam("hello")
assert(s.type == "char *")

s = Spam(f)
assert(s.type == "Foo *")

s = Spam(b)
assert(s.type == "Bar *")

s = Spam(v)
assert(s.type == "void *")

free_void(v)
